extensions/warp-blink-wrtc/src/simple_webrtc/mod.rs

```diff
@@ -479,6 +481,53 @@ impl Controller {
             },
         ));
 
+        let tx = self.event_ch.clone();
+        let dest = peer_id.clone();
+        peer.connection
+            .on_data_channel(Box::new(move |d: Arc<RTCDataChannel>| {
+                let tx2 = tx.clone();
+                let dest2 = dest.clone();
+                d.on_close(Box::new(move || {
+                    if let Err(e) = tx2.send(EmittedEvents::DataChannelClosed {
+                        peer: dest2.clone(),
+                    }) {
+                        log::error!(
+                            "failed to send data channel closed event for peer {}: {}",
+                            &dest2,
+                            e
+                        );
+                    }
+                    Box::pin(async {})
```

Comment:
does the whole return a future? If so maybe we could use `futures::future::ready` instead to wrap the whole scope in it just to make it more idiomatic.